src/protocols/ethereum/libs/EthereumAdapter.ts

```diff
@@ -321,26 +324,82 @@ export class EthereumAdapter extends BaseProtocolAdapter {
     return normalized;
   }
 
-  override async fetchTransactions(
-    address: string,
-    nextPageParams?: string | null,
+  override async fetchAccountTransactions(
+    address: AccountAddress,
+    { nextPageNum }: ITransactionApiPaginationParams = {},
   ): Promise<IFetchTransactionResult> {
-    const { ethActiveNetworkPredefinedSettings } = useEthNetworkSettings();
-    const apiUrl = ethActiveNetworkPredefinedSettings.value.middlewareUrl;
-    const regularTransactions = await new EtherscanService(apiUrl)
-      .fetchAccountTransactions(address, nextPageParams || undefined);
+    const paginationParams: ITransactionApiPaginationParams = {};
+    let regularTransactions: ITransaction[] = [];
+
+    try {
+      const { ethActiveNetworkPredefinedSettings } = useEthNetworkSettings();
+      const service = new EtherscanService(ethActiveNetworkPredefinedSettings.value.middlewareUrl);
+      const [coinTransactions, tokenTransactions] = await Promise.all([
+        service.fetchAccountCoinTransactions(address, { page: nextPageNum }),
+        service.fetchAccountTokenTransactions(address, { page: nextPageNum }),
+      ]);
+
+      // Remove duplicate coin transactions (e.g.: token transfer fee payed with coin)
+      tokenTransactions.forEach((tokenTransaction) => {
+        const index = coinTransactions.findIndex(
+          (coinTransaction) => coinTransaction.hash === tokenTransaction.hash,
+        );
+        if (index > -1) {
+          coinTransactions.splice(index, 1);
+        }
+      });
+
+      regularTransactions = [...coinTransactions, ...tokenTransactions];
+
+      if (regularTransactions?.length) {
+        paginationParams.nextPageNum = ((nextPageNum) ? +nextPageNum + 1 : 2).toString();
```

Comment:
Why if there is no (or `0`) `nextPageNum` value you are setting next page as `2`?

Reply:
Because if we are fetching page without any number it is the 1st page. So the next page is `2`.
So the logic is as follows:
- No `pageNum` => next page = 2
- `pageNum` is set => next page is `pageNum + 1`